lib/zmq-kernel.js

```diff
@@ -331,16 +300,36 @@ export default class ZMQKernel extends Kernel {
       return;
     }
 
-    const { msg_type } = message.header;
+    // input_request messages are attributable to particular execution requests,
+    // and should pass through the middleware stack to allow plugins to see them
+    const { msg_id } = message.parent_header;
+    let callback;
+    if (msg_id) {
+      callback = this.executionCallbacks[msg_id];
+    }
 
-    if (msg_type === "input_request") {
-      const { prompt } = message.content;
+    if (callback) {
+      callback(message, "stdin");
+    } else {
+      // This branch of code should technically never run, but we keep it just
```

Comment:
@n-riesco So you think we should drop this code now, instead of waiting for a few releases?

I guess I thought a deprecation-notice approach will catch if there are any non-compliant kernels out there that would break as a result of removing this functionality. (In the `master` branch, all input requests are replied to, even if they are not in response to an execute request.)

Reply:
Is this a hypothetical question? Are you aware of any kernels setting the wrong `msg_id` in `input_request`?

To my understanding this is a bug in the kernel, and I wouldn't like Hydrogen to corrupt the spec of `input_request` any further (`input_reply` messages are already inconsistent with the rest of the Jupyter protocol: in my opinion, `input_reply` should set `parent_header` as it's done for all the other reply messages).

Reply:
I'm not aware of any kernels doing this. I'll push a commit to get rid of these lines.

Reply:
Side note: there's a race condition if the very first line of Python code you run in hydrogen is `input()`. It looks unrelated to this PR, but worth looking into at some point.

Reply:
@nikitakit 

> there's a race condition if the very first line of Python code you run in hydrogen is input(). 

I don't see the race. Isn't `this.executionCallbaks[msg_id]` created in the same method that sends the `input()` execution request (and thus, before any replies)? [Here](https://github.com/nikitakit/hydrogen/blob/a00cb945eef6b37f8a73b2f0e4422ddaccceb70c/lib/zmq-kernel.js#L226).

Reply:
I have no idea because I can't trigger it anymore. But I swear it happened a few times in a row.

My best guess is that when a new kernel is created and a blue "Hydrogen Kernels updated:" box appears, the input box never gets rendered. But I can't reproduce anymore.

Reply:
So, how about leaving the log message so that we can debug the race the next time it happens?

Reply:
It's completely unrelated to this code change.

Here's a simple issue:
* Run `input()` in Python
* Wait for input prompt to appear
* Activate the command panel using cmd-shift-p
* Watch the input prompt go away permanently

I think the race is probably with some graphics-related event (similar to the command panel activation via cmd-shift-p)

Reply:
Would you open an issue for it (so we don't forget)?